src/common/chunk_copies_calculator.cc

```diff
@@ -97,10 +97,8 @@ void ChunkCopiesCalculator::optimize() {
 		                                       target_slice.size());
 
 		Goal::Slice result(target_slice.getType());
-		i = 0;
-		for (auto result_part : result) {
-			result_part = target_slice[assignment[i]];
-			++i;
+		for (int k = 0, m = target_slice.size(); k < m; ++k) {
```

Comment:
Why not `k < target_slice.size()`? I'm guessing it's because target_slice is not thread safe, but I wanna ask anyway.

Reply:
Applid.